crates/bevy_animation/src/lib.rs

```diff
@@ -0,0 +1,275 @@
+use bevy_transform::{prelude::Transform, TransformSystem};
+use bevy_utils::{tracing::warn, HashMap};
+
+#[allow(missing_docs)]
+pub mod prelude {
+    #[doc(hidden)]
+    pub use crate::{
+        AnimationClip, AnimationPlayer, AnimationPlugin, EntityPath, Keyframes, VariableCurve,
+    };
+}
+
+/// List of keyframes for one of the attribute of a [`Transform`].
+#[derive(Clone, Debug)]
+pub enum Keyframes {
+    /// Keyframes for rotation.
+    Rotation(Vec<Quat>),
+    /// Keyframes for translation.
+    Translation(Vec<Vec3>),
+    /// Keyframes for scale.
+    Scale(Vec<Vec3>),
+}
+
+/// Describes how an attribute of a [`Transform`] should be animated.
+///
+/// `keyframe_timestamps` and `keyframes` should have the same length.
+#[derive(Clone, Debug)]
+pub struct VariableCurve {
+    /// Timestamp for each of the keyframes.
+    pub keyframe_timestamps: Vec<f32>,
+    /// List of the keyframes.
+    pub keyframes: Keyframes,
+}
+
+/// Path to an entity, with [`Name`]s. Each entity in a path must have a name.
+#[derive(Clone, Debug, Hash, PartialEq, Eq, Default)]
+pub struct EntityPath {
+    /// Parts of the path
+    pub parts: Vec<Name>,
+}
+
+/// A list of [`VariableCurve`], and the [`EntityPath`] to which they apply.
+#[derive(Clone, TypeUuid, Debug, Default)]
+#[uuid = "d81b7179-0448-4eb0-89fe-c067222725bf"]
+pub struct AnimationClip {
+    curves: HashMap<EntityPath, Vec<VariableCurve>>,
+    duration: f32,
+}
+
+impl AnimationClip {
+    #[inline]
+    /// Hashmap of the [`VariableCurve`]s per [`EntityPath`].
+    pub fn curves(&self) -> &HashMap<EntityPath, Vec<VariableCurve>> {
+        &self.curves
+    }
+
+    /// Add a [`VariableCurve`] to an [`EntityPath`].
+    pub fn add_curve_to_path(&mut self, path: EntityPath, curve: VariableCurve) {
+        // Update the duration of the animation by this curve duration if it's longer
+        self.duration = self
+            .duration
+            .max(*curve.keyframe_timestamps.last().unwrap_or(&0.0));
+        self.curves.entry(path).or_default().push(curve);
+    }
+}
+
+/// Animation controls
+#[derive(Component, Reflect)]
+#[reflect(Component)]
+pub struct AnimationPlayer {
+    paused: bool,
+    repeat: bool,
+    speed: f32,
+    elapsed: f32,
+    animation_clip: Handle<AnimationClip>,
+}
+
+impl Default for AnimationPlayer {
+    fn default() -> Self {
+        Self {
+            paused: false,
+            repeat: false,
+            speed: 1.0,
+            elapsed: 0.0,
+            animation_clip: Default::default(),
+        }
+    }
+}
+
+impl AnimationPlayer {
+    /// Start playing an animation, resetting state of the player
+    pub fn play_animation(&mut self, handle: Handle<AnimationClip>) -> &mut Self {
+        *self = Self {
+            animation_clip: handle,
+            ..Default::default()
+        };
+        self
+    }
+
+    /// Set the animation to repeat
+    pub fn repeat(&mut self) -> &mut Self {
+        self.repeat = true;
+        self
+    }
+
```

Comment:
```suggestion

    /// Stop the animation from repeating
    pub fn stop_repeating(&mut self) -> &mut Self {
        self.repeat = false;
        self
    }

```
Not sure if my doc comment is even correct english :grimacing: